FlowCrypt/Functionality/Services/Remote Pub Key Services/PubLookup.swift

```diff
@@ -23,12 +33,32 @@ class PubLookup: PubLookupType {
     }
 
     func lookup(with email: String) async throws -> RecipientWithSortedPubKeys {
-        let wkdResult = try await wkd.lookupEmail(email)
-        if !wkdResult.isEmpty {
-            return RecipientWithSortedPubKeys(email: email, keyDetails: wkdResult)
+        let results: [Result] = try await withThrowingTaskGroup(of: Result.self) { tg in
+            var results: [Result] = []
+            tg.addTask {
+                Result(keys: try await self.wkd.lookupEmail(email), source: .wkd)
+            }
+            tg.addTask {
+                Result(keys: try await self.attesterApi.lookupEmail(email), source: .attester)
+            }
+            for try await result in tg {
+                results.append(result)
+            }
+            return results
         }
-
-        let attesterResult = try await attesterApi.lookupEmail(email: email)
-        return RecipientWithSortedPubKeys(email: email, keyDetails: attesterResult)
+        guard let wkdResult = results.first(where: { $0.source == .wkd }) else {
+            throw AppErr.unexpected("expecting to find wkdResult")
+        }
+        guard let attesterResult = results.first(where: { $0.source == .attester }) else {
+            throw AppErr.unexpected("expecting to find attesterResult")
+        }
```

Comment:
Do we need to have both - `wkdResult` and `attesterResult` or only one of them will be enough?

Reply:
Since we put both into task group, and waited for whole task group to finish, we expect both to come out.

Of course, maybe only one of them actually found public key, or none, or both - that's a separate concern.

Reply:
Ah, I see - it won't be `nil` even when keys not found, only `keys` property will be empty and this code will work well.
Previously I thought it'll fail if no keys were found, so all good here 👍

Reply:
I don't like this code pattern much. I've used it twice in this PR. I'm doing tasks in parallel that are not 100% interchangeable, so I need to make sure they all come out and I need to distinguish them (wkd keys have higher importance than attester keys). Maybe there's another code pattern to use? Without having to guard / filter / throw for each task group result.